File: Review-04/Example-6/src/examples/play_tictactoe.py
#! /usr/bin/env python3

from player import (Player, REFERENCE_CYLON)
from game import Game

def main():

    tom = Player("Thomas")
    jay = Player("Jay")

    game = Game(tom, jay)

    # while(!(game.getBoard().isFull())) {
    while game.isNotOver():
        game.playRound()

    print(game.getBoard())

    if game.endedWithWin():
        print("Congratulations {}!".format(game.getWinner()))


if __name__ == "__main__":
    main()

    # @todo add exception handling

File: 15-Java-Strategy-Pattern/Example-1/src/main/java/edu/odu/cs/cs330/examples/PlayTicTacToe.java
package edu.odu.cs.cs330.examples;

import java.io.IOException;

/**
 * The wrapper for main function.
 */
public class PlayTicTacToe
{
    /**
     * This program accepts no command line arguments.
     *
     * @param args not used
     */
    public static void main(String[] args)
        throws IOException
    {
        Game game = Game.builder()
            .withPlayer(new Player("Thomas"))
            .withPlayer(new Player("Jay"))
            .build();

        //while(!(game.getBoard().isFull())) {
        while (game.isNotOver()) {
            game.playRound();
        }

        System.out.println(game.getBoard());

        if (game.endedWithWin()) {
            System.out.printf("Congratulations %s!%n", game.getWinner());
        }
    }
}
